planetiler-core/src/main/java/com/onthegomap/planetiler/util/YAML.java

```diff
@@ -33,12 +36,48 @@ public static <T> T load(Path file, Class<T> clazz) {
   public static <T> T load(InputStream stream, Class<T> clazz) {
     try (stream) {
       Object parsed = snakeYaml.loadFromInputStream(stream);
+      handleMergeOperator(parsed);
       return convertValue(parsed, clazz);
     } catch (IOException e) {
       throw new UncheckedIOException(e);
     }
   }
 
+  /**
+   * SnakeYaml doesn't handle the <a href="https://yaml.org/type/merge.html">merge operator</a> so manually post-process
+   * the parsed yaml object to merge referenced objects into the parent one.
+   */
+  private static void handleMergeOperator(Object parsed) {
+    if (parsed instanceof Map<?, ?> map) {
+      Object toMerge = map.remove("<<");
```

Comment:
Actually, when I read the [merge operator draft](https://yaml.org/type/merge.html) again, it does not prohibit placing the merge operator at any location in a map. Neither does it prohibit multiple instances of merge operators in the same map.

Here is a YAML example where the use of the merge operator is not restricted, as in this suggestion:
```yaml
- base: &base
    nested_mapping: &base_nested_mapping
      key1: value1
      nested_sequence:
        - item1
        - item2
- extension: &extension
    nested_mapping:
      <<: *base_nested_mapping
      key2: value2
      nested_sequence:
        - item3
        - item4
- merged_using_array:
    <<: [ *extension, *base]
- merged_separately:
    <<: *extension
    <<: *base
- merged_interleaved:
    start: marker1
    <<: *extension
    mid: marker2
    <<: *base
    end: marker3
```
According to [YAML Lint](https://www.yamllint.com/), it would expand to
```yaml
- base:
    nested_mapping:
      key1: value1
      nested_sequence:
        - item1
        - item2
- extension:
    nested_mapping:
      key1: value1
      nested_sequence:
        - item3
        - item4
      key2: value2
- merged_using_array:
    nested_mapping:
      key1: value1
      nested_sequence:
        - item3
        - item4
      key2: value2
- merged_separately:
    nested_mapping:
      key1: value1
      nested_sequence:
        - item3
        - item4
      key2: value2
- merged_interleaved:
    start: marker1
    nested_mapping:
      key1: value1
      nested_sequence:
        - item3
        - item4
      key2: value2
    mid: marker2
    end: marker3
```
Please feel free to reject this suggestion

Reply:
Yeah I noticed that playing around with pyyaml. Since this approach happens post-parsing and the parser throws a clear exception in this case and you can easily switch to `<<: [*a, *b]` I think it's an OK limitation.  A more fully-thought-out merge operator might support things like merging maps in the order they are specified when the operator appears more than once (and working for lists) but that would require support at the parser level.